testnet/cfgsync/src/repo.rs

```diff
@@ -76,36 +77,33 @@ impl ConfigRepo {
     async fn run(&self) {
         let timeout_duration = self.timeout_duration;
 
-        match timeout(timeout_duration, self.wait_for_hosts()).await {
-            Ok(_) => {
-                println!("All hosts have announced their IPs");
-
-                let mut waiting_hosts = self.waiting_hosts.lock().unwrap();
-                let hosts = waiting_hosts
-                    .iter()
-                    .map(|(host, _)| host)
-                    .cloned()
-                    .collect();
-
-                let configs = create_node_configs(
-                    self.consensus_params.clone(),
-                    self.da_params.clone(),
-                    self.tracing_settings.clone(),
-                    hosts,
-                );
-
-                for (host, sender) in waiting_hosts.drain() {
-                    let config = configs.get(&host).expect("host should have a config");
-                    let _ = sender.send(RepoResponse::Config(Box::new(config.to_owned())));
-                }
+        if timeout(timeout_duration, self.wait_for_hosts()).await == Ok(()) {
```

Comment:
```suggestion
        if timeout(timeout_duration, self.wait_for_hosts()).await.is_ok() {
```

Reply:
Again I think in this case is better to match on the expected result inside the `Ok()`. Pedantic Clippy wont' complain, but the restriction lints I am working on actually will do. If the `timeout` function changes for a reason, I think it would be the least to understand "why", and take action based on that. Using `is_ok()` basically assumes the result is `()` or something useless, but this assumption might not hold true forever in the future. What do you think?